File: models/song.go
package models

type Song struct {
	Id       int    `json:"id" db:"id"`
	Filename string `json:"filename" db:"filename"`
	Artist   string `json:"artist" db:"artist"`
	Title    string `json:"title" db:"title"`
	Format   string `json:"format" db:"format"`
}

File: gui/src/sdk/requests/song.ts
export type COVER_SOURCE = 'NO_COVER'|'LINK'|'UPLOADED';
export type FORMAT = 'CDG'|'WEBM'|'MP4';

export type SongPost = {
    title: string;
    artist: string;
    format: string;
    cover_type: COVER_SOURCE;
    cover_data: string;

    song: FileList;
    cdg: FileList;
};